targets/unity-v2/source/Authentication/PlayFabAuthService.cs

```diff
@@ -0,0 +1,228 @@
+        GameCenter,
+        Kongregate,
+        NintendoSwitch,
+        XBoxLive,
+        PlayStation,
+        OpenId,
+        WindowsHello,
+        Twitch,
+        Steam
+    }
+
+    public class PlayFabAuthService
+    {
+        // Events to subscribe to for this service
+        public delegate void DisplayAuthenticationEvent();
+        public event DisplayAuthenticationEvent OnDisplayAuthentication;
+
+        public delegate void LoginSuccessEvent(LoginResult success);
+        public event LoginSuccessEvent OnLoginSuccess;
+
+        public delegate void PlayFabErrorEvent(PlayFabError error);
+        public event PlayFabErrorEvent OnPlayFabError;
+
+        public delegate void PlayFabLink(AuthTypes authType, PlayFabError error = null);
+        public event PlayFabLink OnPlayFabLink;
+        public event PlayFabLink OnPlayFabUnlink;
+
+        public string Email;
+        public string Username;
+        public string Password;
+        public GetPlayerCombinedInfoRequestParams InfoRequestParams;
+        public bool ForceLink;
+        
+        public string AuthTicket;
+        public string OpenIdConnectionId;
+        public string WindowsHelloChallengeSignature;
+        public string WindowsHelloPublicKeyHint;
+        
+        public string PlayFabId { get; private set; }
+        public string SessionTicket { get; private set; }
+        public static EntityKey EntityToken { get; private set; }
+        public PlayFabAuthenticationContext AuthenticationContext { get; private set; }
+
+        private const string _LoginRememberKey = "PlayFabLoginRemember";
+        private const string _PlayFabRememberMeIdKey = "PlayFabIdPassGuid";
+        private const string _PlayFabAuthTypeKey = "PlayFabAuthType";
+
+        public bool RememberMe
+        {
+            get { return PlayerPrefs.GetInt(_LoginRememberKey, 0) != 0; }
+            set { PlayerPrefs.SetInt(_LoginRememberKey, value ? 1 : 0); }
+        }
+
+        /// <summary>
+        /// Remember the type of authenticate for the user
+        /// </summary>
+        public AuthTypes AuthType
+        {
+            get { return (AuthTypes) PlayerPrefs.GetInt(_PlayFabAuthTypeKey, 0); }
+            set { PlayerPrefs.SetInt(_PlayFabAuthTypeKey, (int) value); }
+        }
+
+        /// <summary>
+        /// Generated Remember Me ID
+        /// Pass Null for a value to have one auto-generated.
+        /// </summary>
+        public string RememberMeId
+        {
+            get { return PlayerPrefs.GetString(_PlayFabRememberMeIdKey, ""); }
+            set
+            {
+                var guid = string.IsNullOrEmpty(value) ? Guid.NewGuid().ToString() : value;
+                PlayerPrefs.SetString(_PlayFabRememberMeIdKey, guid);
+            }
+        }
+
+        private static readonly Dictionary<AuthTypes, IAuthenticationStrategy> AuthStrategies = new Dictionary<AuthTypes, IAuthenticationStrategy>();
+
+        static PlayFabAuthService()
```

Comment:
We should move the contents of this to the instance constructor, rather than the static one.